gdx/src/com/badlogic/gdx/maps/tiled/BaseTmjMapLoader.java

```diff
@@ -0,0 +1,746 @@
+			}
+			idToObject.put(id, object);
+			objects.add(object);
+		}
+	}
+
+	private void loadProperties (final MapProperties properties, JsonValue element) {
+		if (element == null) return;
+
+		if (element.name() != null && element.name().equals("properties")) {
+			for (JsonValue property : element) {
+				final String name = property.getString("name", null);
+				String value = property.getString("value", null);
+				String type = property.getString("type", null);
+				if (value == null) {
+					value = property.asString();
+				}
+				if (type != null && type.equals("object")) {
+					// Wait until the end of [loadTiledMap] to fetch the object
+					try {
+						// Value should be the id of the object being pointed to
+						final int id = Integer.parseInt(value);
+						// Create [Runnable] to fetch object and add it to props
+						Runnable fetch = new Runnable() {
+							@Override
+							public void run () {
+								MapObject object = idToObject.get(id);
+								properties.put(name, object);
+							}
+						};
+						// [Runnable] should not run until the end of [loadTiledMap]
+						runOnEndOfLoadTiled.add(fetch);
+					} catch (Exception exception) {
+						throw new GdxRuntimeException(
+							"Error parsing property [\" + name + \"] of type \"object\" with value: [" + value + "]", exception);
+					}
+				} else {
+					Object castValue = castProperty(name, value, type);
+					properties.put(name, castValue);
+				}
+			}
+		}
+	}
+
+	private Object castProperty (String name, String value, String type) {
+		if (type == null) {
+			return value;
+		} else if (type.equals("int")) {
+			return Integer.valueOf(value);
+		} else if (type.equals("float")) {
+			return Float.valueOf(value);
+		} else if (type.equals("bool")) {
+			return Boolean.valueOf(value);
+		} else if (type.equals("color")) {
+			// Tiled uses the format #AARRGGBB
+			String opaqueColor = value.substring(3);
+			String alpha = value.substring(1, 3);
+			return Color.valueOf(opaqueColor + alpha);
+		} else {
+			throw new GdxRuntimeException(
+				"Wrong type given for property " + name + ", given : " + type + ", supported : string, bool, int, float, color");
+		}
+	}
+
+	private Cell createTileLayerCell (boolean flipHorizontally, boolean flipVertically, boolean flipDiagonally) {
+		Cell cell = new Cell();
+		if (flipDiagonally) {
+			if (flipHorizontally && flipVertically) {
+				cell.setFlipHorizontally(true);
+				cell.setRotation(Cell.ROTATE_270);
+			} else if (flipHorizontally) {
+				cell.setRotation(Cell.ROTATE_270);
+			} else if (flipVertically) {
+				cell.setRotation(Cell.ROTATE_90);
+			} else {
+				cell.setFlipVertically(true);
+				cell.setRotation(Cell.ROTATE_270);
+			}
+		} else {
+			cell.setFlipHorizontally(flipHorizontally);
+			cell.setFlipVertically(flipVertically);
+		}
+		return cell;
+	}
+
+	protected void loadTileSet (JsonValue element, FileHandle tmjFile, ImageResolver imageResolver) {
```

Comment:
There were parsing issues in the loadTileSet() method. As it was this code would only work with .tmj files with embedded tilesets. Which is why it worked fine with the SuperKoalio test map as it used an embedded tileset.

I fixed up the method to make sure it works with tilesets that are kept in the .tsj format, external to the tilemap .tmj file.
Fully working method and updated method below:

```
protected void loadTileSet (JsonValue element, FileHandle tmjFile, ImageResolver imageResolver) {
		  if (element.getString("firstgid") != null) {
				int firstgid = element.getInt("firstgid", 1);
				String imageSource = "";
				int imageWidth = 0;
				int imageHeight = 0;
				FileHandle image = null;

				String source = element.getString("source", null);
				if (source != null) {
					 FileHandle tsj = getRelativeFileHandle(tmjFile, source);
					 try {
						  element =json.parse(tsj);
						  if (element.has("image")) {
								imageSource = element.getString("image");
								imageWidth = element.getInt("imagewidth", 0);
								imageHeight = element.getInt("imageheight", 0);
								image = getRelativeFileHandle(tsj, imageSource);
						  }
					 } catch (SerializationException e) {
						  throw new GdxRuntimeException("Error parsing external tileSet.");
					 }
				} else {
					 if (element.has("image")) {
						  imageSource = element.getString("image");
						  imageWidth = element.getInt("imagewidth", 0);
						  imageHeight = element.getInt("imageheight", 0);
						  image = getRelativeFileHandle(tmjFile, imageSource);
					 }
				}
				String name = element.getString("name", null);
				int tilewidth = element.getInt("tilewidth", 0);
				int tileheight = element.getInt("tileheight", 0);
				int spacing = element.getInt("spacing", 0);
				int margin = element.getInt("margin", 0);

				JsonValue offset = element.get("tileoffset");
				int offsetX = 0;
				int offsetY = 0;
				if (offset != null) {
					 offsetX = offset.getInt("x", 0);
					 offsetY = offset.getInt("y", 0);
				}
				TiledMapTileSet tileSet = new TiledMapTileSet();

				// TileSet
				tileSet.setName(name);
				final MapProperties tileSetProperties = tileSet.getProperties();
				JsonValue properties = element.get("properties");
				if (properties != null) {
					 loadProperties(tileSetProperties, properties);
				}
				tileSetProperties.put("firstgid", firstgid);

				// Tiles
				JsonValue tiles = element.get("tiles");

				if (tiles == null) {
					 tiles = new JsonValue(JsonValue.ValueType.array);
				}

				addStaticTiles(tmjFile, imageResolver, tileSet, element, tiles, name, firstgid, tilewidth, tileheight, spacing, margin,
					source, offsetX, offsetY, imageSource, imageWidth, imageHeight, image);

				Array<AnimatedTiledMapTile> animatedTiles = new Array<>();

				for (JsonValue tileElement : tiles) {
					 int localtid = tileElement.getInt("id", 0);
					 TiledMapTile tile = tileSet.getTile(firstgid + localtid);
					 if (tile != null) {
						  AnimatedTiledMapTile animatedTile = createAnimatedTile(tileSet, tile, tileElement, firstgid);
						  if (animatedTile != null) {
								animatedTiles.add(animatedTile);
								tile = animatedTile;
						  }
						  addTileProperties(tile, tileElement);
						  addTileObjectGroup(tile, tileElement);
					 }
				}
				// replace original static tiles by animated tiles
				for (AnimatedTiledMapTile animatedTile : animatedTiles) {
					 tileSet.putTile(animatedTile.getId(), animatedTile);				}

				map.getTileSets().addTileSet(tileSet);

		  }
	 }
```